### PR TITLE
Use the displayname in lost password emails where possible

### PR DESCRIPTION
**Pull request by tomneedham**
## Description
Use displayname in email to fields where possible, nicer than the uid

## Motivation and Context
Looks nicer.

## How Has This Been Tested?
Manually sending lost password mails and looking at mailhog.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: core/Controller/LostController.php [hunk 1]
```diff
@@ -258,14 +258,15 @@ public function setPassword($token, $userId, $password, $proceed) {
 	protected function sendNotificationMail($userId) {
 		$user = $this->userManager->get($userId);
 		$email = $user->getEMailAddress();
+		$name = $user->getDisplayName() === false ? $userId : $user->getDisplayName();
```
Comment (PVince81) [inline core/Controller/LostController.php]: you should also use the use id if the display name is empty string or spaces for some reason ? might be safer

in any case, I winced from the idea that display name can be `false`, and not `null`

**Review by PVince81 — COMMENTED** [review]
see comment, otherwise this looks good

**Comment by tomneedham** [comment]
No - it will auto fallback to the UID. My bad. Updated the PR.

**Comment by PVince81** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase failed! Please rebase your pull request manually via the command line.

Reason:
```
Command: git rebase base/master

Exit Code: 128

Cause:

.git/rebase-apply/patch:278: trailing whitespace.
	Scenario: Change full name 
.git/rebase-apply/patch:302: trailing whitespace.
	Scenario: Change full name 
warning: 2 lines add whitespace errors.
error: Failed to merge in the changes.


```

**Comment by sharidas** [comment]
@phil-davis @individual-it A helping hand here would be nice. Because of the failure https://drone.owncloud.com/owncloud/core/15141/604, I tried to add a trivial change to the feature file. But that's not the problem, if my understanding is correct. Looks like the iteration is not happening at https://github.com/owncloud/core/blob/master/tests/TestHelpers/EmailHelper.php#L91 or say its not entering the loop.

**Comment by phil-davis** [comment]
@sharidas there was a 500 server error:
```
[Thu Feb 14 10:53:00 2019] Undefined variable: userObject at /home/phil/git/owncloud/core/core/Controller/LostController.php#348
[Thu Feb 14 10:53:00 2019] Exception: {"Exception":"Error","Message":"Call to a member function getDisplayName() on null","Code":0,"Trace":"#0 \/home\/phil\/git\/owncloud\/core\/core\/Controller\/LostController.php(209): OC\\Core\\Controller\\LostController->sendEmail('qqqq', 'KAc9GFGvFhSzjmy...', 'http:\/\/172.17.0...')\n#1 \/home\/phil\/git\/owncloud\/core\/lib\/private\/AppFramework\/Http\/Dispatcher.php(153): OC\\Core\\Controller\\LostController->email('qmail@abc.com')\n#2 \/home\/phil\/git\/owncloud\/core\/lib\/private\/AppFramework\/Http\/Dispatcher.php(85): OC\\AppFramework\\Http\\Dispatcher->executeController(Object(OC\\Core\\Controller\\LostController), 'email')\n#3 \/home\/phil\/git\/owncloud\/core\/lib\/private\/AppFramework\/App.php(100): OC\\AppFramework\\Http\\Dispatcher->dispatch(Object(OC\\Core\\Controller\\LostController), 'email')\n#4 \/home\/phil\/git\/owncloud\/core\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php(47): OC\\AppFramework\\App::main('LostController', 'email', Object(OC\\AppFramework\\DependencyInjection\\DIContainer), Array)\n#5 \/home\/phil\/git\/owncloud\/core\/lib\/private\/Route\/Router.php(342): OC\\AppFramework\\Routing\\RouteActionHandler->__invoke(Array)\n#6 \/home\/phil\/git\/owncloud\/core\/lib\/base.php(906): OC\\Route\\Router->match('\/lostpassword\/e...')\n#7 \/home\/phil\/git\/owncloud\/core\/index.php(54): OC::handleRequest()\n#8 {main}","File":"\/home\/phil\/git\/owncloud\/core\/core\/Controller\/LostController.php","Line":348}
[Thu Feb 14 10:53:00 2019] 192.168.1.12:45630 [500]: /index.php/lostpassword/email
```
I pushed a commit to fix it.
It is good that we have tests!

**Comment by sharidas** [comment]
Review required for this PR. Let me know if this looks ok.

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
I don't get why the code coverage would decrease as the change isn't affecting any old/new code paths.
And as we can't open the report :man_shrugging:

**Comment by PVince81** [comment]
@sharidas please backport

**Comment by sharidas** [comment]
Backport PR https://github.com/owncloud/core/pull/34512

**Pull request merged**